inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -113,45 +107,60 @@ def article_exists(obj, eng):
     return False
 
 
-@with_debug_logging
-def belongs_to_relevant_category(obj, eng):
-    """Return True if the record is Core or Non-Core.
+def auto_approve(obj, eng):
+    """Check if auto approve the current ingested article.
 
     Arguments:
         obj: a workflow object.
         eng: a workflow engine.
 
-    Returns:
-        bool: ``True`` if the ingested article belongs to a relevant
-        category, else ``False``.
-
+    Return:
+        bool: True when the record belongs to an arXiv category that is fully
+        harvested or if the primary category is `physics.data-an`, otherwise
+        False.
     """
-    return get_coreness(obj.data) is not Coreness.non_relevant
+    return has_fully_harvested_category(obj.data) or \
+           physics_data_an_is_primary_category(obj.data)
 
 
-def get_coreness(record):
-    """Return the coreness of the record harvested.
+def has_fully_harvested_category(record):
+    """Check if the record in `obj.data` has fully harvested categories.
 
-    Args:
-        record(dict): the record harvested.
+    Arguments:
+        obj: a workflow object.
+        eng: a workflow engine.
 
     Return:
-        a Coreness value in according to the record's coreness level.
+        bool: True when the record belongs to an arXiv category that is fully
+        harvested, otherwise False.
     """
-    arxiv_categories = set(get_arxiv_categories(record))
-    relevant_categories = current_app.config.get('ARXIV_CATEGORIES', [])
-
-    if arxiv_categories & set(relevant_categories['core']):
-        return Coreness.core
-    elif arxiv_categories & set(relevant_categories['non-core']):
-        return Coreness.non_core
-    else:
-        return Coreness.non_relevant
+    record_categories = set(get_arxiv_categories(record))
+    harvested_categories = current_app.config.get('ARXIV_CATEGORIES', {})
+    return len(
+        record_categories &
+        set(
+            harvested_categories.get('core') +
+            harvested_categories.get('non-core')
+        )
+    ) > 0
+
+
+def physics_data_an_is_primary_category(record):
+    record_categories = set(get_arxiv_categories(record))
```

Comment:
if you don't cast to a `set` here, you can reuse the variable later.

Reply:
copy/paste leftover :+1: